Refactor test method names and add doc comments

koriym opened the pull request:
**+phpstan 2 and vimeo 6 support**

Align method docblock with detailed type annotations for improved clarity and stricter type safety. Add a  definition and Psalm suppression for better static analysis support.

* + `#[Override]` https://wiki.php.net/rfc/marking_overriden_methods
* + final keyword default https://psalm.dev/docs/running_psalm/issues/ClassMustBeFinal/

---

This pull request enhances the type safety annotations in CacheReader by aligning the method docblock with detailed type annotations. This improvement enhances code clarity and enforces stricter type safety measures. Additionally, a definition and Psalm suppression have been added to enhance static analysis support.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit


- **Chores**
  - Updated continuous integration to use PHP 8.4.
  - Upgraded development and static analysis tool dependencies.
  - Expanded code quality checks.

- **Refactor**
  - Enhanced type safety and in-code documentation.
  - Enforced stricter class inheritance controls with final declarations.
  - Applied minor formatting and clean-up improvements.

- **Tests**
  - Renamed and documented test methods for better clarity.
  - Adjusted assertions to improve validation and maintainability.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

koriym committed:
Update dependencies to latest versions in composer.lock

Upgraded multiple packages, including `amphp/amp` to v3.1.0 and related dependencies for compatibility with PHP 8.1. Added several new libraries such as `amphp/cache`, `amphp/parallel`, and `amphp/pipeline`. These updates improve overall functionality, compatibility, and performance of the project.

koriym committed:
Add #[\Override] attribute to applicable methods

This commit uniformly applies the #[\Override] attribute to methods that override parent class methods or implement interfaces. This change improves code clarity by explicitly indicating method overrides, aiding in code maintenance and readability.

koriym committed:
Make classes final for improved immutability and safety.

Marking classes as final prevents unintended extension, ensuring stricter adherence to design principles and better maintainability. This change reduces risks of unforeseen behavior and increases code clarity.

koriym committed:
Add phpstan/phpstan-phpunit to composer dependencies

This change includes the phpstan/phpstan-phpunit package in the vendor-bin/tools/composer.json file. It enhances PHPUnit support for static analysis, improving code quality and test reliability.

koriym committed:
Refactor and improve type hints, assertions, and annotations

Replaced inline `assert` calls with `PHPUnit` assertions for better test readability and robustness. Improved type hints and added missing annotations for clarity and consistency across files. Removed unused statements and comments to simplify the codebase.

koriym committed:
Soothe CS: Replace #[\Override] attributes with #[Override]

Unified the usage of attribute syntax across the codebase by replacing #[\Override] with #[Override]. This improves code consistency and adheres to the preferred attribute naming conventions.

koriym committed:
Update PHP version to 8.4 in workflows

Updated the PHP version from 8.3 to 8.4 in both coding standards and static analysis GitHub workflows. This ensures compatibility and aligns with the latest PHP version support.

koriym committed:
Add Psalm template annotations to CacheReader method

Introduce @template and @param annotations to improve static analysis

koriym committed:
Refactor docblock to adjust template annotation.

Moved the template annotation below the param tag for better structure and consistency with coding standards. No functional changes introduced.

koriym committed:
Refactor test method names and add detailed doc comments

Renamed test methods for improved clarity and consistency, aligning them with their respective functionalities. Added detailed doc comments to better describe each test's purpose, enhancing code readability and maintainability.

koriym committed:
Enhance type safety annotations in CacheReader

Aligned the method docblock with detailed type annotations to improve clarity and enforce stricter type safety. Added a template definition and Psalm suppression for better static analysis support.

koriym commented:
@coderabbitai review
@sourcery-ai review

NaokiTsuchiya commented on src/Matcher.php:
```diff
@@ -60,6 +65,7 @@ public function startsWith($prefix): AbstractMatcher
     /**
      * {@inheritdoc}
      */
+    #[\Override]
```
```suggestion
    #[Override]
```

`use Override;` しているので、 `\` は不要でしょうか？

koriym replied:
cs-fixの対象から漏れてましたね。対応しました！

NaokiTsuchiya commented on src/Matcher.php:
```diff
@@ -68,6 +74,7 @@ public function logicalOr(AbstractMatcher $matcherA, AbstractMatcher $matcherB)
     /**
      * {@inheritdoc}
      */
+    #[\Override]
```
```suggestion
    #[Override]
```

koriym committed:
Fix attribute syntax and update code style checks

Replaced incorrect attribute syntax `[\Override]` with `[Override]` for consistency. Added 'sl-src' directory to PHP CodeSniffer configuration to ensure style checks cover the new directory.

koriym committed:
Add Override.php to autoload files in composer.json

The Override.php class was added and registered in the autoload section of composer.json. This ensures the class is loaded automatically if not already defined,(=PHP 8.0-8.2)

koriym commented:
@NaokiTsuchiya Attribute Readerなどでインスタンス化されてしまったりするとクラスが無いことでエラーになってしまうことを防ぐために[2d78490](https://github.com/ray-di/Ray.Aop/pull/241/commits/2d784903bf9b54539a4d39d1ed7e4b8daef1fc75)を加えました。

koriym committed:
Remove deprecated `#[Override]` attribute and related class.

The `#[Override]` attribute and its usages have been removed across the codebase, as it is no longer required. The `src-compat/Override.php` file has also been deleted, and references in `composer.json` were updated to reflect the change.

koriym committed:
Set MissingOverrideAttribute to suppress in psalm.xml

Updated psalm configuration to suppress MissingOverrideAttribute errors. This ensures the codebase avoids unnecessary warnings for missing override attributes, streamlining the static analysis process.

koriym committed:
Update Psalm configuration to target PHP 8.4

Upgraded the PHP version in the Psalm configuration file from 8.3 to 8.4. This aligns the static analysis tool with the project's updated PHP version requirements.

koriym commented:
Closing this PR as it's superseded by #244. The changes in this PR are either:
1. Already included in #244 (#[Override] attributes)
2. No longer relevant (CacheReader improvements - now deprecated in src-deprecated/)
3. Redundant (PHP 8.4 support already in CI)

#244 provides a more comprehensive update with the migration to PHP 8 attributes.
